xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/internal/AbstractAsynchronousEventStore.java

```diff
@@ -439,37 +439,41 @@ private <O, I> void complete(EventStoreTask<O, I> task, O output)
         task.future.complete(output);
 
         // Notify event listeners
+        Object notificationOuput = task.output;
+        if (task.output instanceof Optional<?>) {
+            notificationOuput = ((Optional<?>) task.output).orElse(null);
```

Comment:
I'm not sure it's enough.

We should probably not send any notification if the optional is not present (meaning nothing was in fact deleted).